tests/write_tests.cpp

```diff
@@ -102,6 +103,169 @@ TEST_CASE("Try writing a glTF with all buffers and images", "[write-tests]") {
 	REQUIRE(std::filesystem::exists(exportedFolder / "image1.bin"));
 }
 
+static bool run_official_gltf_validator(const std::filesystem::path &model_path) {
+	if (!std::filesystem::exists(gltfValidator)) {
+		return false;
+	}
+	// run the validator on the model
+	std::string command = std::string(gltfValidator) + " -m " + model_path.string();
+	auto result = std::system(command.c_str());
+	if (result != 0) {
+		return false;
+	}
+	return true;
+}
+
+static void inline test_glb_rewrite(const std::filesystem::path &model_parent_dir, const std::filesystem::path &model_file, bool skip_official_validator = false) {
+	if (!skip_official_validator && !std::filesystem::exists(gltfValidator)) {
+		SKIP("Failed to find the official glTF validator (download and unzip to tests/gltf/gltf_validator)");
+		return;
+	}
+
+	if (!skip_official_validator && !run_official_gltf_validator(sampleAssets / model_parent_dir / model_file)) {
+		auto report = std::string( "Failed to validate the model: " + (model_parent_dir / model_file).string());
+		SKIP(report.c_str());
+		return;
+	}
+	
+	auto sampleDir = sampleAssets / model_parent_dir;
+	auto samplePath = sampleDir / model_file;
+	fastgltf::GltfFileStream sampleFileStream(samplePath);
+	REQUIRE(sampleFileStream.isOpen());
+	static constexpr auto requiredExtensions = static_cast<fastgltf::Extensions>(~0U);
+
+
+	//std hashset
+	std::unordered_set<std::string_view> requiredExtensionsSet;
+	for (const auto& [name, value] : fastgltf::extensionStrings) {
+		requiredExtensionsSet.insert(name);
+		REQUIRE((requiredExtensions & value) == value);
+	}
+	fastgltf::Parser parser(requiredExtensions);
+	auto options = fastgltf::Options::LoadExternalBuffers | fastgltf::Options::LoadExternalImages ;
+	fastgltf::Expected<fastgltf::Asset> asset = model_file.extension() == ".gltf" ?
+		parser.loadGltfJson(sampleFileStream, sampleDir, options) :
+		parser.loadGltfBinary(sampleFileStream, sampleDir, options);
+	REQUIRE(getErrorName(asset.error()) == "None");
+	REQUIRE(getErrorName(fastgltf::validate(asset.get())) == "None");
```

Comment:
Why are you doing this?

Reply:
I wanted to get the actual error name, but since we don't return any validate errors other than "InvalidGLTF", it's pointless